Major update

steffilazerte opened the pull request:
Fixes some issues introduced by the update to lubridate v.0.7.0 (now v.0.7.1). Adds some functionality and tweaks code for rOpenSci application. Also added codecoverage and allowed the R Devel tests to fail without reporting a total failure on Travis CI and AppVeyor.

There's also a draft of the paper for JOSS (too long?)

Would almost suggest going straight to master with this, so we can get on the rOpenSci application soon? I probably should have pushed some of these fixes to the devel branch long ago!

In Detail: 

## Changes
- Updated `stations` data
- Added `flags` and `glossary` datasets as well as vignettes
- `stations_search()` warns user if name looks like coords
- `stations_search()` with `coord` now returns closest 10 stations
- `add_weather()` warns user if trying to interpolate weather from >1 station
- Updated code to conform with rOpenSci requirements
- Data downloaded from multiple timezones defaults to UTC

## Bug fixes
- `weather(format = FALSE)` properly returns data
- updated `weather()` to work with `lubridate` 1.7.1

boshek commented on R/stations.R:
```diff
@@ -22,20 +22,28 @@
-#' @import magrittr
 #' @export
 
 stations_all <- function(url = "http://bit.ly/2sIuCty",
```
This function still fails locally for me. 
```
> stations_all()
Error in stations_all() : 
  'url' must point to a csv file either local or online.
In addition: Warning message:
In file(con, "r") :
  cannot open URL 'http://bit.ly/2sIuCty': HTTP status was '301 Moved Permanently'
```

steffilazerte commented:
#### Re Sam's changes

- I added the date in the DESCRIPTION file so the citation information would be correct. Hopefully if we're accepted by JOSS we can use that citation in the future. But if you prefer no date, it doesn't worry me.

- I don't think I like putting the README under tools. According to [Writting R Extensions](https://cran.r-project.org/doc/manuals/r-release/R-exts.html#Data-in-packages) (just above the link location) "Subdirectory tools is the preferred place for auxiliary files needed during configuration, and also for sources need to re-create scripts (e.g. M4 files for autoconf). " And I don't think the README needs to be installed, shouldn't it be fine to just put it and it's figure folder into the `.Rbuildignore` file?. If you think it really should be installed, I think we should put it in the `inst` folder, not the tools folder.

#### Re Sam's comments/questions

- I don't mind keeping the version at  0.2.2.9000 assuming we go to 0.2.3 after rOpenSci (
34e2b65 )

- I'm fine to sign releases, will look into that for the next one.

- sysdata.rda is the internal data frame that contains some of the dataframes used to assign column names etc. it's created [here](https://github.com/steffilazerte/weathercan/blob/c4fe1b0b0ea3f662ba400746a8c337d05ec223e2/data-raw/data-index.R#L11)

- I've added a fix so that `stations_all` works fine on windows. The original version for me, and passed the tests on Travis CI and AppVeyor, so I don't really understand what's happening. On my windows computer I got the same problem, but not when I used the real link, ie:

```
stations_all(url = "ftp://client_climate@ftp.tor.ec.gc.ca/Pub/Get_More_Data_Plus_de_donnees/Station%20Inventory%20EN.csv")
```

Perhaps it was the shortened link? I didn't really like using a shortened link to begin with but I was following Hadley's suggestion for fixing extra long lines in the documentation (http://r-pkgs.had.co.nz/check.html). So I got rid of the long link in the documentation but the script now uses it directly (
063aaf8)

#### Conclusions
- Let me know what your opinion is regarding the README files and I'll move them around

boshek commented:
> And I don't think the README needs to be installed

What do you mean by that? CRAN builds README files locally so they have to runnable. Therefore this applies:

> "Subdirectory tools is the preferred place for auxiliary files needed during configuration, and also for sources need to re-create scripts (e.g. M4 files for autoconf). "

I interpret that because README has to be built with pandoc on CRAN they want any images in the tools folder. FWIW, this is a new policy. Here is the message from CRAN last March (though I can't find the original):

> It was recently pointed out to us that some README.html files (generated
> from the corresponding README.md ones) on the CRAN package web pages are
> incomplete, missing 'local' images not available from the web page and
> in most cases actually not even shipped with the package. This clearly
> should be changed, so we will move to using '--self-contained' for the
> pandoc conversion to ensure that the README.html files are "complete".
> 
> Of course, this implies that all 'local' images used in README.md are
> needed in the package sources.
> 
> If the images are also used for vignettes or Rd files, you can put them
> in the 'vignettes' or 'man/figures' directories. Otherwise, please put
> them in the top-level 'tools' directory, or a subdirectory of it.
> 
> The CRAN incoming checks in r-devel were changed to perform the pandoc
> conversion checks with '--self-contained', and hence will warn about
> missing images.
> 
> Pls ensure completeness in the next regular update of your package.

I just removed the date from DESCRIPTION because nothing has been release yet. I typically do that as a last step. So no problem with the date itself. 

Everything else looks great.

steffilazerte commented:
Okay that makes sense, thanks for clarifying. I think I've always thought of the README as something for github, but it is relevant to the package too.
